restutil/processor.go

```diff
@@ -0,0 +1,707 @@
+					_, err = io.WriteString(os.Stdout, "\n")
+				}
+				if err != nil {
+					select {
+					case errs <- err:
+					default:
+					}
+					return
+				}
+			}
+		}()
+	}
+
+	wg := sync.WaitGroup{}
+
+	for i := 0; i < conns; i++ {
+		wg.Add(1)
+		go func() {
+			if err := rp.putAll(docs, failChan); err != nil {
+				select {
+				case errs <- err:
+				default:
+				}
+			}
+			wg.Done()
+		}()
+	}
+
+	for {
+		var doc map[string]interface{}
+		if err := dec.Decode(&doc); err != nil {
+			if err == io.EOF {
+				break
+			}
+			return err
+		}
+		select {
+		case docs <- doc:
+		case err := <-errs:
+			return err
+		}
+	}
+
+	close(docs)
+
+	wg.Wait()
+
+	if dumpFailed {
+		close(failChan)
+		failwg.Wait()
+	}
+
+	select {
+	case err := <-errs:
+		return err
+	default:
+		return nil
+	}
+
+}
+
+func DiffIDs(sourceURL, destURL string) error {
+	sourceIDs := make(chan *string)
+	go fetchIDList(sourceURL, sourceIDs)
+
+	destIDs := make(chan *string)
+	go fetchIDList(destURL, destIDs)
+
+	sources := make(map[string]struct{})
+	dests := make(map[string]struct{})
+
+	for sourceIDs != nil || destIDs != nil {
+		select {
+		case sourceID, ok := <-sourceIDs:
+			if !ok {
+				sourceIDs = nil
+			} else {
+				sources[*sourceID] = struct{}{}
+			}
+		case destID, ok := <-destIDs:
+			if !ok {
+				destIDs = nil
+			} else {
+				dests[*destID] = struct{}{}
+			}
+		}
+	}
+
+	var output struct {
+		OnlyInSource      []string `json:"only-in-source"`
+		OnlyInDestination []string `json:"only-in-destination"`
+	}
+
+	output.OnlyInSource = []string{}
+	output.OnlyInDestination = []string{}
+
+	for s := range sources {
+		if _, found := dests[s]; !found {
+			output.OnlyInSource = append(output.OnlyInSource, s)
+		} else {
+			delete(dests, s)
+		}
+
+	}
+
+	for s := range dests {
+		output.OnlyInDestination = append(output.OnlyInDestination, s)
+	}
+
+	return json.NewEncoder(os.Stdout).Encode(output)
+
+}
+
+type SyncService struct {
+	SourceIDsRetriever IDListRetriever
+	DestIDsRetriever   IDListRetriever
+	SourceURL          string
+	DestURL            string
+	MaxConcurrentReqs  int
+	MinExecTime        int
+	Retries            int
+	Deletes            bool
+}
+
+func SyncIDs(service *SyncService) error {
+	errChan := make(chan error)
+	defer close(errChan)
+	sourceIDs := make(chan string)
+	go service.SourceIDsRetriever.Retrieve(sourceIDs, errChan)
+	destIDs := make(chan string)
+	go service.DestIDsRetriever.Retrieve(destIDs, errChan)
+
+	sources := make(map[string]struct{})
+	dests := make(map[string]struct{})
+
+	for sourceIDs != nil || destIDs != nil {
+		select {
+		case sourceID, ok := <-sourceIDs:
+			if !ok {
+				sourceIDs = nil
+			} else {
+				sources[sourceID] = struct{}{}
+			}
+		case destID, ok := <-destIDs:
+			if !ok {
+				destIDs = nil
+			} else {
+				dests[destID] = struct{}{}
+			}
+		case err := <-errChan:
+			return err
+		}
+	}
+
+	var output struct {
+		Created int `json:"created"`
+		Deleted int `json:"created"`
```

Comment:
I didn't touch those bits... they were there before I moved `processor.go` to the new directory. Annoyingly this makes the pull request harder to read :(